### PR TITLE
Refactors disposal bins

### PR DESCRIPTION
**Pull request by XDTM**
Splits the mode var into more readable and independent vars.
Related to #25304; it doesn't fix it outright since the power determines if it's accumulating pressure and it is technically off when the bin is already full. It will, however, look off when it's full now, and if pressed it will turn on then off, which makes it less confusing.

### CODE REVIEW

**Comment by XDTM** [comment]
Should i remove those files from the commit or try to fix the conflict?

**Comment by Cyberboss** [comment]
If you didn't touch the keys used in ui_data or ui_act, just remove them

**Comment by XDTM** [comment]
i did touch them

**Comment by Cyberboss** [comment]
merge upstream, then recompile, then use your versions

**Comment by XDTM** [comment]
thanks

File: code/modules/recycling/disposal-unit.dm [hunk 1]
```diff
@@ -12,8 +12,10 @@
 	obj_integrity = 200
 	max_integrity = 200
 	resistance_flags = FIRE_PROOF
+	interact_open = TRUE
 	var/datum/gas_mixture/air_contents	// internal reservoir
-	var/mode = PRESSURE_ON
+	var/full_pressure = TRUE
```
Comment (AnturK) [inline code/modules/recycling/disposal-unit.dm]: Shouldn't this start with false to preserve old functionality ?

Comment (Cyberboss) [inline code/modules/recycling/disposal-unit.dm]: It should tbh

**Pull request merged**